### ISSUE TITLE
 Increasing latency on Android and IOS notifications, problem with the push server?

### ISSUE THREAD

**Issue by TBI-NS** [nextcloud/spreed#9759]
Hello to the entire team,

Today at 17:45 French time (which is 11:45 AM Eastern Time in the USA), we noticed once again delays on the ANDROID IOS notifications of our talk. We had a similar problem a few months ago, the end result being that the Nextcloud server, which is responsible for routing to the Apple and Google servers, was saturated, making it virtually impossible to receive notifications.

I have found the server at https://push-notifications.nextcloud.com/ and have taken the liberty to put it under monitoring to ensure that it does not consistently go down, but today's new delays lead me to believe that we are facing the same issue with this server causing our slowdowns.

Could we have more transparency about this server? Is it well protected against DDoS attacks? Is there load balancing in place? I only find one IP, 95.217.53.153 with a ping of 35ms, as a service behind the subdomain push-notifications.nextcloud.com, and I find this concerning.

Thank you for any information you can provide to the Nextcloud user community. As of today, our server operates with 100 users, all on talk, and it has become vital for us to be able to communicate without loss!

**Comment by TBI-NS** [comment]
Update : The four NextCloud instances that I manage are now all affected by a complete outage of notifications. The instances are not in the same data centers or locations, which confirms that it is indeed the push server that is at fault.

**Comment by SystemKeeper** [comment]
Hello TBI-NS,

this sounds like a environment better suited for Nextcloud Enterprise than a community version. If you're a customer I'd advice to reach out to support, if you would like to become one please contact sales.

**Comment by TBI-NS** [comment]
Hello, we are not a business but a non-profit association under the 1901 law in France, we do not have a for-profit business model and do not make any financial profit from the use of NextCloud. So, is your answer to my question simply that I have to pay? When I find myself tweeting one of the team's developers on his personal account in the middle of the weekend, and the service miraculously starts working again 20 minutes after my message, what should I understand? That I didn't get a thank you or that it's pure coincidence? 

For your information, a VPS at OVH cloud located in France costs less than ten euros per month... With all the assurance that their data center provides, do you really find it serious to answer me in this way? If I were given the choice to contribute by making a donation, my answer would be yes 👍, but I'm not going to pay for a business service when I chose NextCloud because it's open source. 

I use Nextcloud and support the project as best I can, but if this is indeed your official answer, with all due respect, it seems to me to be laxity concerning this push service. The Nextcloud application suite suits me perfectly for all my uses, the only service I depend on is your PUSH server. Why not propose us to run one ourselves?

Just a reminder, the last time this service had a bug, it was only because you forgot to rotate your logs and you saturated the storage... How to explain this when you are one of the big names in the dev field?

Thank you for all.

**Comment by nickvergessen** [comment]
> or that it's pure coincidence?

Mostly this. Our system does not record any downtime for the 10th June
![grafik](https://github.com/nextcloud/spreed/assets/213943/9cae1291-0c92-4cb4-aad4-5a7053a2c9d4)

Also the response time of the testing endpoint is in full average on that day.

> we noticed once again delays on the ANDROID IOS notifications of our talk

Mind to extend that a bit? How long was the delay?

**Comment by TBI-NS** [comment]
Hello nickvergessen, I thank you for reopening the discussion on a ticket that has been closed. 
I misspoke about the latency, so I will attempt to provide as much information as possible.

Indeed, I did not mean to say that notifications globally have too much latency, in general the delays are acceptable.

I am surprised to see an uptime of 100% on your push service because my nextcloud is also being monitored and it reports a 100% uptime, the day I wrote this ticket we were all impacted, notifications were arriving 1 out of 10 times or not at all (probably depending on the number of people in the channel, typically if the message was sent in a two-person conversation the notification arrived with a lot of delay, but in a group of 80 people it was completely broken.) The issue simultaneously affected my nextcloud, but also a friend's which is hosted in a different datacenter than mine, so I ruled out a breakdown on our side.

Moreover, the problem was on both android and IOS so I immediately ruled out the possibility of an issue on either side with their push transmission services. (If I understand correctly, Google and Apple enforce the use of a push system from their servers)

That leaves only one possible issue.. the NC push service.

Currently everything is back to normal and notifications arrive with a maximum delay of 4 to 6 seconds which is convincing and sufficient.

I want to specify that throughout the entire period when the notifications were no longer coming in, we were able to discuss without any problems, the NC was responding perfectly and the HPB + Turn did not report any abnormal logs.

Thank you for your follow-up, I remain at your disposal if you want to dig deeper.

Best regards
